mps-sync-plugin-lib/src/main/kotlin/org/modelix/mps/sync/transformation/modelixToMps/incremental/ModelixTreeChangeVisitor.kt

```diff
@@ -256,11 +284,17 @@ class ModelixTreeChangeVisitor(
                 return@enqueue null
             }
 
-            logger.error { "We missed a move to new parent case for Modelix Node ($nodeId)." }
+            val message = "A containment changed case for Node ($nodeId) was missed."
+            notifyAndLogError(message)
 
             null
         }
     }
 
     private fun getNode(nodeId: Long) = replicatedModel.getBranch().getNode(nodeId)
+
+    private fun notifyAndLogError(message: String) {
+        val exception = ModelixToMpsSynchronizationException(message)
+        notifierInjector.notifyAndLogError(message, exception, logger)
```

Comment:
Consider making the `error` in `fun notifyAndLogError(message: String, error: Throwable, logger: KLogger)` optional. Here an exception created just to call the error method.

Reply:
Not really. The purpose of the Exception is to show that the error occurred while synchronizing from Modelix to MPS. The direction is important as it helps the us to identify when the error occurred. See message prefix in the ModelixToMpsSynchronizationException: `Error occurred, while synchronizing to the server:`

Reply:
Then why not just prepend the message with `Error occurred, while synchronizing to the server:`?
(My rule of thumb is: Exceptions should only be used, when something needs to be thrown.)

Reply:
My points of instantiating the Exceptions are:
1. Prepend the prefix at one place.
2. Capture the fact that something bad happened while we were doing the synchronization (in either direction).
3. Capture the cause of the exception if there was such.
4. Have a stack trace that lead to that error. It is useful in logging, because there we do `logger.error(ex)` which prints the stack trace by default. If we do not wrap the `message` into an Exception, then we lose this traceability.

Therefore, I would like to continue instantiating the wrapper exception here, in the Node/Model/ModuleTransformer and in the Model/ModuleSynchronizer (for the opposite direction).

--------

Some weaker arguments of why I do the prepending in the constructors:

Because `ModelixToMpsSynchronizationException` is instantiated at 6 places (across 5 files),  and `MpsToModelixSynchronizationException` is instantiated at 4 places (across 4 files).

That would mean at least 10 places where I have to prepend the same constant to the message. Having the two constructs instead of them reduces a lot of duplicates in the code.

Reply:
> 4. Have a stack trace that lead to that error.

Ah ok, that is the only thing, you can not really get without using Exceptions.

Though, I would prefer messages that have enough information to understand the problem and unambiguous identify the invalid state without the stacktrace, I am fine with exceptions.